search_service/proxy/atlas.py

```diff
@@ -45,71 +35,187 @@ def __init__(self, *,
+            db_name = qn.get('db_name', '')
+            db_cluster = qn.get('cluster_name', '')
+
+            tags: List[Tag] = []
+
+            for classification in entity.classificationNames or list():
+                tags.append(Tag(tag_name=classification))
+
+            badges: List[Tag] = tags
+
+            table = Table(name=entity_name,
+                          key=f"{entity.typeName}://{db_cluster}.{db_name}/{entity_name}",
+                          description=entity_attrs.get('description'),
                           cluster=db_cluster,
-                          database=table.typeName,
+                          database=entity.typeName,
                           schema=db_name,
-                          column_names=columns,
                           tags=tags,
                           badges=badges,
-                          last_updated_timestamp=table_attrs.get('updateTime'))
+                          column_names=[],
+                          last_updated_timestamp=entity_attrs.get('updateTime'))
+
+            result.append(table)
+
+        return result
+
+    def _atlas_basic_search(self, query_params: Dict) -> Tuple[List[Table], int]:
+        """
+        Conduct search using Atlas Basic Search API.
 
-            table_results.append(table)
+        :param query_params: A dictionary of query parameters needed to be pass to Basic Search Post method of Atlas
+        :return: List of Table objects and approximate count of entities matching in Atlas
+        """
 
-        return table_results
+        try:
+            # Fetch the table entities based on query terms
+            search_results = self.atlas.search_basic.create(data=query_params)
+        except BadRequest as ex:
+            LOGGER.error(f"Fetching Tables Failed : {str(ex)}")
+            return [], 0
+
+        if not len(search_results.entities):
+            return [], 0
+
+        # noinspection PyProtectedMember
+        results_count = search_results._data.get("approximateCount")
+
+        results = self._prepare_tables(search_results.entities, enhance_metadata=False)
+
+        return results, results_count
+
+    def _prepare_basic_search_query(self, limit: int, page_index: int, query_term: Optional[str] = None,
+                                    filters: Optional[List[Tuple[str, str, str]]] = None,
+                                    operator: Optional[str] = None,
+                                    classifications: Optional[List[str]] = None,
+                                    entity_type: str = None) -> Dict[str, Any]:
+        """
+        Render a query for Atlas Basic Search API.
+
+        :param query_term: Search Query Term
+        :param limit:
+        :param page_index:
+        :param fitlers: Optional list of tuples (field, condition, value) that will translate into entityFilters for
+        narrowing down search results
+        :param operator:
```

Comment:
nit: Supported values for operator?

Reply:
good catch, I will make this one general effort to work on docs